### PR TITLE
issue86 macfix

### LINKED ISSUE #86: New Localdev image fails to build on MacOS with error on `groupmod`
```
Step 18/29 : RUN     groupmod -g $DOCKER_GID docker &&     groupadd -g $GID -o $UNAME &&     useradd -d /home/$UNAME -g $GID -m -s /bin/bash -u $UID $UNAME &&     mkdir /workspace &&     chown $UNAME:$UNAME /workspace
 ---> Running in eb6200041fdc
groupmod: GID '20' already exists
2022/11/05 16:11:45 Failed to create container localdev-status: Error response from daemon: No such image: localdev-server:latest
```

### PR DESCRIPTION
**Pull request by gamerson**
- New Localdev image fails to build on MacOS with error on groupmod
- issue86 TEST ONLY
- change owner of docker.sock to localdev

### CODE REVIEW

**Comment by rotty3000** [comment]
I believe the fix should look more like the windows fix but I'll confirm

**Comment by rotty3000** [comment]
Hey @gamerson with @ipeychev's help we confirmed that it appears your solution is the only one that will work on Mac. However before we merge I want to consolidate the solution on Lin & Win. I'll send some follow up changes once I've tested on those platforms.

File: docker/images/localdev-server/entrypoint.sh [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 set -e
 
+sudo chown localdev /var/run/docker.sock
```
Comment (gamerson) [inline docker/images/localdev-server/entrypoint.sh]: Should this username be passed in as ENV VAR? The same way it is passed in as ARG in dockerfile?

Comment (rotty3000) [reply docker/images/localdev-server/entrypoint.sh]: We'll have to make sure the Dockerfile sets the default value of the ENV var from the arg but I can do that.

**Comment by gamerson** [comment]
I've just tested this, and I'm seeing a new msg that could be a concern
```
Writing Kubernetes config: could not create any of the following paths: /run/user/502/tilt-dev/tilt-default/cluster
█
```

**Comment by rotty3000** [comment]
@gamerson is that a REST uri or a file system path? If it's a FS is that inside the `localdev-server` container or on local system?

**Comment by gamerson** [comment]
> @gamerson is that a REST uri or a file system path? If it's a FS is that inside the `localdev-server` container or on local system?

This is a file system path, inside the `localdev-server`. This is from the `Tiltfile` resource inside of Tilt UI, so its a message from Tilt server itself. 

image

Looks like tilt is trying to serialize out the config for connecting to the kube context and is running into permission error writing to `/run/user/502`. The `/run` path is same as `/var/run` in our localdev server and its owned by root.  I'm not sure why tilt thinks it should be writable.  

One option might be to make `/var/run/user` recusrively owned by our `localdev` user instead of just  `/var/run/docker.sock`.

File: docker/images/localdev-server/entrypoint.sh [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 set -e
 
+sudo chown "$LOCALDEV_USER" /var/run/docker.sock
```
Comment (gamerson) [inline docker/images/localdev-server/entrypoint.sh:5]: We could add `sudo chown -R "$LOCALDEV_USER" /var/run/user` as well.

Comment (rotty3000) [reply docker/images/localdev-server/entrypoint.sh:5]: pushed this fix.

**Comment by gamerson** [comment]
@rotty3000 I'm good with merging this.

**Pull request merged**